apps/season-editor/src/pages/components/sub-competences.tsx

```diff
@@ -0,0 +1,87 @@
+import { Box, Button } from "@mui/material";
+import { grey } from "@mui/material/colors";
+import { StyledInput } from "./styled-input";
+import { useFieldArray, useFormContext } from "react-hook-form";
+import { FormInputs } from "../types";
+
+type SubCompetencesProps = {
+  competenceAreaId: string;
+  competenceId: string;
+};
+
+export const SubCompetences = ({
+  competenceAreaId,
+  competenceId,
+}: SubCompetencesProps) => {
+  const { register, control } = useFormContext<FormInputs>();
+  const { fields: subCompetenceFields, append: appendSubCompetence } =
+    useFieldArray({
+      control,
+      name: "subCompetences",
+    });
+
+  return (
+    <Box
+      sx={{
+        paddingLeft: 4,
+        marginTop: 2,
+        display: "flex",
+        flexDirection: "column",
+        gap: 2,
+        borderLeft: `1px solid ${grey[300]}`,
+      }}
+    >
+      {subCompetenceFields
+        .filter(
+          (subCompetenceField) =>
+            subCompetenceField.competenceAreaId === competenceAreaId &&
+            subCompetenceField.competenceId === competenceId
+        )
+        .map((subCompetenceField, index) => (
+          <Box
+            key={subCompetenceField.id}
+            sx={{
+              display: "flex",
+              gap: 2,
+            }}
+          >
+            <StyledInput short>
+              <label>id</label>
+              <input
+                disabled
+                type="text"
+                {...register(
+                  `subCompetences.${index}.competenceId` as "subCompetences.0.subCompetenceId"
```

Comment:
I think there's a sub missing here ... it also prints the wrong ID when I load the canary season.yaml.

<img width="493" alt="Screenshot 2023-11-27 at 12 56 18" src="https://github.com/talent-digital/talentkit/assets/797483/f60f762a-9d38-4043-8d64-841205cabcb6">

Reply:
outdated